### PR TITLE
SDA-4514: NSIS Migration

### PR DESCRIPTION
**Pull request by NguyenTranHoangSym**
## Description
Adding new NSIS script for migration of config file from 23.1 upto day
this script will eliminate duplicated config

### CODE REVIEW

File: build/installer.nsh [hunk 1]
```diff
@@ -19,6 +60,153 @@ Function uninstallSymphony
...
+FunctionEnd
+
+# latestAutoUpdateChannelEnabled
+Function updateConfigLatestAutoUpdateChannelEnabled
+	push true
+	push latestAutoUpdateChannelEnabled
+	call SearchAndReplace
+FunctionEnd
+
 !macro copySystemGlobalConfig
```
Comment (KiranNiranjan) [inline build/installer.nsh:210]: This logic also needs to be applied for per user install

File: build/installer.nsh [hunk 2]
```diff
@@ -1,4 +1,45 @@
+Var STR_HAYSTACK
+Var STR_NEEDLE
+Var STR_CONTAINS_VAR_1
+Var STR_CONTAINS_VAR_2
+Var STR_CONTAINS_VAR_3
+Var STR_CONTAINS_VAR_4
+Var STR_RETURN_VAR
+ 
+Function StrContains
```
[the remark this answers is not in the document]
Comment (NguyenTranHoangSym) [reply build/installer.nsh:9]: This is a suggestion function coming from NSIS that I might have to copy it. It is not available as default
Here is a reference. I recommend not to modify anything above
https://nsis.sourceforge.io/StrContains

File: build/installer.nsh [hunk 3]
```diff
@@ -56,6 +244,12 @@ FunctionEnd
         !insertmacro UAC_RunElevated
         Quit
     ${endif}
+    call updateConfigIsPodUrlEditable
+	call updateConfigForceAutoUpdate
```
[the remark this answers is not in the document]
Comment (NguyenTranHoangSym) [reply build/installer.nsh:248]: Yep indentation I dont have any format available

File: build/installer.nsh [hunk 4]
```diff
@@ -1,4 +1,45 @@
+Var STR_HAYSTACK
+Var STR_NEEDLE
+Var STR_CONTAINS_VAR_1
```
[the remark this answers is not in the document]
Comment (NguyenTranHoangSym) [reply build/installer.nsh:3]: https://nsis.sourceforge.io/StrContains
Here is a reference to code explanation. This is actually a library

File: build/installer.nsh (hunk 1 above)
Comment (NguyenTranHoangSym) [reply build/installer.nsh:220]: I added the logic to per user

File: build/installer.nsh [hunk 5]
```diff
@@ -19,6 +60,163 @@ Function uninstallSymphony
...
+	#FileOpen $8 "E:\Projects\NSIS\text.txt" w
+	#FileWrite $8 "$0$\r$\n"
+	#FileWrite $8 "$1$\r$\n"
+	#FileWrite $8 "$4$\r$\n"
+	#FileClose $8
+
+	#LogText $0
+	#DetailPrint $0
+	#DetailPrint $1
+	#DetailPrint $4
```
Comment (KiranNiranjan) [inline build/installer.nsh:161]: Remove this?

File: build/installer.nsh [hunk 6]
```diff
@@ -19,6 +60,163 @@ Function uninstallSymphony
...
+					FileWrite $R4 $R1
+					Goto loopFind
+		${EndIf}
+		done:
+			FileClose $R4
+			FileClose $4
+			#FileWrite $8 "At done"
+			#FileWrite $8 "$4$\r$\n"
+			#FileWrite $8 "At done"
+			#FileWrite $8 "$R4$\r$\n"
```
Comment (KiranNiranjan) [inline build/installer.nsh:118]: Remove this?

**Review by KiranNiranjan — APPROVED** [review]
LGTM 👍🏻 remove comments

**Pull request merged**